test/Confluent.Kafka.Transactions/TestProducer.cs

```diff
@@ -0,0 +1,84 @@
+using Confluent.Kafka;
+using System;
+using System.Threading;
+
+
+namespace Confluent.Kafka.Transactions
+{
+    public class TestProducer
+    {
+        static TimeSpan DefaultTimeout = TimeSpan.FromSeconds(30);
+        Random random;
+
+        SimulationConfig conf;
+        int id;
+        string bootstrapServers;
+
+        public TestProducer(string bootstrapServers, int id, SimulationConfig simulationConfig)
+        {
+            this.id = id;
+            this.bootstrapServers = bootstrapServers;
+
+            // ensure different string of pseudo-random behavior for each test producer.
+            random = new Random(simulationConfig.RandomSeedBase + id);
```

Comment:
Randomness is not a great test attribute, since it makes it hard to reproduce failing tests.

Reply:
it's pseudo-random - fully reproducible.